### DISCUSSION TITLE
Refresh a different route.

### DISCUSSION THREAD

**Discussion by ephraimduncan** [vercel/next.js#62641] [Help] [answered]
### Summary

I have a server component dashboard page fetching data directly from the db that displays a list of posts. When you delete a post on a  `/post/[id]` page and then you are redirected back to the `/dashboard` page with `router.push("/dashboard")`, the deleted post still appears in the list until you manually refresh the page in the browser. From the docs of `router.refresh()`

> `router.refresh()`: Refresh the current route...

How do I refresh another route?

### Additional information

```js
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.1.0: Mon Oct  9 21:28:45 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T6020
Binaries:
  Node: 18.17.0
  npm: 10.2.4
  Yarn: 1.22.21
  pnpm: 8.7.5
Relevant Packages:
  next: 14.1.0
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

**Comment by icyJoseph** [comment]
Use, revalidatePath, https://nextjs.org/docs/app/api-reference/functions/revalidatePath:

- Currently, revalidatePath invalidates all the routes in the [client-side Router Cache](https://nextjs.org/docs/app/building-your-application/caching#router-cache). This behavior is temporary and will be updated in the future to apply only to the specific path.

**Reply by ephraimduncan** [reply]
I tried it, it doesn't work. I've been getting this error

```
Error: Invariant: static generation store missing in revalidateTag _N_T_/dashboard
    at revalidateTag (revalidate-tag.js:16:15)
    at revalidatePath (revalidate-path.js:25:45)
    at handleDeleteForm (form-settings.tsx:66:21)
```

For

```js
revalidatePath("/dashboard")
```

**Reply by icyJoseph** [reply]
oh, that does not look good, never seen that error before, it claims the issue is in revalidate tag as well, but you call revalidate path.

**Reply by icyJoseph** [reply]
Found some issues but no concrete pointers.

Are you using `turbo`?

**Reply by ephraimduncan** [reply]
No unfortunately

**Reply by ephraimduncan** [reply, marked as answer]
I fixed it, I was using `revalidatePath` wrongly. I was using it in a client component causing the issue. I created an action and it works.

```ts
"use server";

import { revalidatePath } from "next/cache";

export const refreshDashboardAfterDeletion = async () => {
  revalidatePath("/dashboard");
};
```

**Reply by icyJoseph** [reply]
Ah yeah! those details can be easy to miss... well good job!
